### ISSUE TITLE
Automation API create_or_select_stack ignores work_dir parameter

### ISSUE THREAD

**Issue by felixwang9817** [pulumi/pulumi#8436]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

### Steps to reproduce

This [line](https://github.com/pulumi/pulumi/blob/83e24765f3d3ddc99d26e8785b6c38d13eb4cfe3/sdk/python/lib/pulumi/automation/_local_workspace.py#L434) does not use the `work_dir` parameter, which seems to be a mistake (at the very least, it is inconsistent with the behavior described in the docstring).

Expected: 
Actual:

**Comment by emiliza** [comment]
Thanks for filing @felixwang9817 - this is interesting. It's unclear why `work_dir` isn't explicitly called out there as a top level option, but this does exist in `opts`

**Comment by felixwang9817** [comment]
@emiliza yes, I got the intended result by setting `work_dir` in `opts`. I just wanted to report this since the docs are incorrect, which may confuse folks that don't want to look at the source code.

**Comment by squaremo** [comment]
I think this is a case of misleading API. The *_stack procedures behaves modally, depending on which keyword arguments you supply:

 - if `program` and `project_name` are in the keyword args (see https://github.com/pulumi/pulumi/blob/06ba63bb57e90706c1550861b785075ae860144a/sdk/python/lib/pulumi/automation/_local_workspace.py#L386), the resulting stack refers to an inline program;
 - otherwise if `work_dir` is in the keyword args (see https://github.com/pulumi/pulumi/blob/06ba63bb57e90706c1550861b785075ae860144a/sdk/python/lib/pulumi/automation/_local_workspace.py#L541), the stack refers to a program constructed from that directory.

The `work_dir` _keyword argument_ means "this is where the program is". There's _also_ a field in the `opts` argument called `work_dir`, which means "use this as the current directory". For a local program, the keyword argument overrides the option (https://github.com/pulumi/pulumi/blob/06ba63bb57e90706c1550861b785075ae860144a/sdk/python/lib/pulumi/automation/_local_workspace.py#L552); for an inline program, the `work_dir` keyword argument is ignored and the `work_dir` option is defaulted.

The docstring says (emphasis mine) "unless a `work_dir` **option** is specified" -- so it's accurate, but a bit of a trap.

It would be less confusing if the keyword argument had a different name like `project_dir`. But changing it would break existing code; so I suggest three steps:

 1. be much clearer about what the keyword arguments do or don't do, in the docstring
 2. add a `project_dir` keyword argument to each of the *_stack procedures, which is used in preference to the `work_dir` argument
 3. eventually (at whatever boundary breaking changes are allowed), deprecate the `work_dir` keyword parameter.
